api/v1alpha1/nnfstorageprofile_types.go

```diff
@@ -24,6 +24,22 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// NnfStorageProfileLustreMiscData defines args to pass to mkfs, mount, and other utilities.
+type NnfStorageProfileLustreMiscData struct {
+	// ZpoolCreate specifies the zpool create commandline, minus the "zpool create".
+	// This is where you may specify zpool create options, and the virtual device (vdev) such as
+	// "mirror", or "draid".  See zpoolconcepts(7).
+	ZpoolCreate string `json:"zpoolCreate,omitempty"`
+
+	// Mkfs specifies the mkfs.lustre commandline, minus the "mkfs.lustre".
+	// Use the --mkfsoptions argument to specify the zfs create options.  See zfsprops(7).
+	Mkfs string `json:"mkfs,omitempty"`
+
+	// Mount specifies mount options for the mount-utils library.
+	// Use one array element per option. Do not prepend the options with "-o".
+	Mount []string `json:"mount,omitempty"`
```

Comment:
I think it would be more user friendly if these were all of the same format. Either a string of all the arguments or an array of each argument.